atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/RetryLimitReachedException.java

```diff
@@ -17,19 +17,36 @@
 package com.palantir.atlasdb.keyvalue.api;
 
 import java.util.Arrays;
+import java.util.Collections;
 import java.util.List;
 
 import com.palantir.common.exception.AtlasDbDependencyException;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.SafeLoggable;
 
-public class RetryLimitReachedException extends AtlasDbDependencyException {
-    private static final String MESSAGE = "Request was retried %d times and failed each time for the request.";
+public class RetryLimitReachedException extends AtlasDbDependencyException implements SafeLoggable {
+    private static final String MESSAGE = "Request was retried and failed each time for the request.";
+
+    private final int numRetries;
 
     public RetryLimitReachedException(List<Exception> exceptions) {
-        super(String.format(MESSAGE, exceptions.size()));
+        super("");
```

Comment:
I thought we normally preserve messages? e.g. as in Remoting's `SafeIoException` , `InvalidAcceptorCacheKeyException` in Atlas

```suggestion
        super(MESSAGE);
```